### PR TITLE
Pass the id of the server to remove to event listeners

### LINKED ISSUE #854: Destroying a server sometimes destroys the wrong server
https://youtu.be/ZQ3mkNRaAZA

Console output printing the selected server to be deleted:
```
{
  "id": "238362934",
  "dropletInfo": {
    "id": 238362934,
    "name": "OutlineServerNewYork",
     ...
    ],
  ...
  "serverConfig": {
    "name": "Outline Server New York",
    "serverId": "9e98b0a0-6fc6-4262-99a4-c80cddf9379c",
    "metricsEnabled": false,
    "createdTimestampMs": 1616599153759,
    "version": "1.6.0",
  }
}
```

Showing that the wrong server was chosen to be destroyed by the event handler

### PR DESCRIPTION
**Pull request by JonathanDCohen**
Fix #854

Tested:
Successfully added and removed both manual and managed servers correctly.

### CODE REVIEW

File: src/server_manager/web_app/app.ts [hunk 1]
```diff
@@ -1072,9 +1072,9 @@ export class App {
         });
   }
 
-  private deleteSelectedServer() {
-    const serverToDelete = this.selectedServer;
-    const serverId = serverToDelete.getId();
+  private deleteServer(serverId: string) {
+    const serverToDelete = this.idServerMap.get(serverId);
+    console.error(serverToDelete);
```
Comment (fortuna) [inline src/server_manager/web_app/app.ts:1077]: Remove

File: src/server_manager/web_app/app.ts [hunk 2]
```diff
@@ -1091,7 +1091,7 @@ export class App {
           .then(
               () => {
                 this.removeServer(serverId);
-                this.appRoot.selectedServer = null;
+                this.appRoot.selectedServerId = '';
                 this.selectedServer = null;
```
Comment (fortuna) [inline src/server_manager/web_app/app.ts:1095]: `removeServer` already takes care of updating the selected server.
```suggestion
```

File: src/server_manager/web_app/app.ts [hunk 3]
```diff
@@ -1107,9 +1107,8 @@ export class App {
     });
   }
 
-  private forgetSelectedServer() {
-    const serverToForget = this.selectedServer;
-    const serverId = serverToForget.getId();
+  private forgetSelectedServer(serverId: string) {
+    const serverToForget = this.idServerMap.get(serverId);
```
Comment (fortuna) [inline src/server_manager/web_app/app.ts:1111]:
```suggestion
    const serverToForget = this.getServerById(serverId);
```

**Review by fortuna — CHANGES_REQUESTED** [review]
Nice, thanks for the quick fix

File: src/server_manager/web_app/app.ts (hunk 1 above)
Comment (JonathanDCohen) [reply src/server_manager/web_app/app.ts]: Done in 1059f04

**Review by fortuna — APPROVED** [review]
Let's make sure it's working after the latest changes.
Thanks again for the quick fix.

**Pull request merged**